### PR TITLE
feat/keadm: support customsized labels when use join command

### PR DESCRIPTION
**Pull request by ttlv**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
we want to use keadm to support customsized labels  
**Which issue(s) this PR fixes**:

Fixes #
https://github.com/kubeedge/kubeedge/issues/2764

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go [hunk 1]
```diff
@@ -122,6 +123,19 @@ func (ku *KubeEdgeInstTool) createEdgeConfigFiles() error {
...
+		s1 := strings.Replace(ku.Labels, "{", "", -1)
+		s2 := strings.Replace(s1, "}", "", -1)
+		labels := strings.Split(s2, ",")
+		for _, label := range labels {
+			key := strings.Split(label, ":")[0]
+			value := strings.Split(label, ":")[1]
+			labelsMap[key] = value
+		}
+		edgeCoreConfig.Modules.Edged.Labels = labelsMap
+	}
```
Comment (Iceber) [inline keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go:138]: If the value of the label tag is like `{"key1":"value1", "key2":"value2"}`,then you can use the `json.Unmarshal`
```go
        if strings.TrimSpace(ku.Labels) != "" {
                labels := map[string]string{}
                if err := json.Unmarshal([]byte(ku.Labels), &labels); err != nil {
                        return fmt.Errorf("invalid Labels: %s", err)
                }
                edgeCoreConfig.Modules.Edged.Labels = labels
        }
```

But I recommend using `StringSliceVarP` to resolve the label flag

File: keadm/cmd/keadm/app/cmd/edge/join.go [hunk 2]
```diff
@@ -114,6 +114,9 @@ func addJoinOtherFlags(cmd *cobra.Command, joinOptions *types.JoinOptions) {
 
 	cmd.Flags().StringVar(&joinOptions.TarballPath, types.TarballPath, joinOptions.TarballPath,
 		"Use this key to set the temp directory path for KubeEdge tarball, if not exist, download it")
+
+	cmd.Flags().StringVarP(&joinOptions.Labels, types.Labels, "l", joinOptions.Labels,
+		`use this key to set the customized labels for node. you can input customized labels like {"key1":"value1", "key2":"value2"}`)
```
Comment (Iceber) [inline keadm/cmd/keadm/app/cmd/edge/join.go:119]: I recommend using StringSliceVarP to resolve the label flag, like `-l key1=value1,key2=value2`.
What do you think?

Comment (ttlv) [reply keadm/cmd/keadm/app/cmd/edge/join.go]: lgtm,thanks for your suggesstion

File: keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go (hunk 1 above)
Comment (ttlv) [reply keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go:134]: thanks for your suggesstion

**Comment by ttlv** [comment]
@fisherxu this pr is ready to get lgtm now :)

File: keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go [hunk 3]
```diff
@@ -122,6 +123,15 @@ func (ku *KubeEdgeInstTool) createEdgeConfigFiles() error {
 	}
 	edgeCoreConfig.Modules.EdgeStream.TunnelServer = net.JoinHostPort(cloudCoreIP, strconv.Itoa(constants.DefaultTunnelPort))
 
+	if len(ku.Labels) >= 1 {
+		labelsMap := make(map[string]string)
+		for _, label := range ku.Labels {
+			key := strings.Split(label, "=")[0]
+			value := strings.Split(label, "=")[1]
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go:130]: In the flag comment it's `:` to separate the key and value, here is `=`, which one does the user should chose?

Comment (ttlv) [reply keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go:130]: yeap I have fixed description and use '=' to replace ':'.I'm sorry for that :)

Comment (fisherxu) [reply keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go:130]: lgtm, please fix the DCO issue :)

Comment (ttlv) [reply keadm/cmd/keadm/app/cmd/util/edgecoreinstaller.go:130]: ok,DCO had been fixed

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**